File: src/1-500/019/RemoveNth.py
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def removeNthFromEnd(self, head: ListNode, n: int) -> ListNode:
        root = ListNode(0)
        root.next = head
        fast, slow, pre = root, root, root
        while n - 1:
            fast = fast.next
            n -= 1
        while fast.next:
            fast = fast.next
            pre = slow
            slow = slow.next
        pre.next = slow.next
        return root.next


solution = Solution()
head = ListNode(1)
p = head
for i in range(2, 10):
    node = ListNode(i)
    p.next = node
    p = node

solution.removeNthFromEnd(head, 3)
p = head
while p:
    print(p.val)
    p = p.next

File: src/1-500/019/RemoveNth.go

type ListNode struct {
    Val int
 	Next *ListNode
}

func endReached(head *ListNode, n int) bool {
	
	for probe != nil || n > 0 {
		probe = probe.Next
		n--
	}
	return (probe == nil && n == 0)
}
 
func removeNthFromEnd(head *ListNode, n int) *ListNode {
	var curr *ListNode = head	
	var probe *ListNode = head
	for curr != nil {
		if endReached(curr,n) {

		} else {
			curr = curr.Next
		}
	}
    
}